scene/resources/material.cpp

```diff
@@ -1062,12 +1062,12 @@ void BaseMaterial3D::_update_shader() {
 		code += "	TANGENT = vec3(0.0,0.0,-1.0) * abs(NORMAL.x);\n";
 		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.y);\n";
 		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.z);\n";
-		code += "	TANGENT = normalize(TANGENT);\n";
+		code += "	TANGENT = normalize(TANGENT) * MODEL_NORMAL_MATRIX;\n";
```

Comment:
The extra code should only be added when using world triplanar on UV1 (world triplanar on UV2 doesn't matter because UV2 is never used for normal mapping)

Reply:
alright, I'll add this conditional as soon as I find the opportunity :)